lib/archethic.ex

```diff
@@ -54,32 +56,37 @@ defmodule Archethic do
   @doc """
   Send a new transaction in the network to be mined. The current node will act as welcome node
   """
-  @spec send_new_transaction(Transaction.t()) :: :ok
+  @spec send_new_transaction(Transaction.t(), Crypto.key(), nil | Contract.Context.t()) :: :ok
```

Comment:
I'm wondering if we should not move this function into the `Mining` context, so nested modules under `lib/archethic` will not call the top level `Archethic` but the siblings' context APIs.

Reply:
Referenced in #1106